optimum/amd/cli.py

```diff
@@ -1,47 +1,111 @@
 import os
+import subprocess
 import sys
-from argparse import ArgumentParser
 
-from .topology_utils import extract_max_avg_bandwidth_cluster, get_bandwidth_matrix
 
+try:
+    subprocess.run(["rocm-smi"], check=True)
 
-def amdrun():
+    ROCM_AVAILABLE = True
+except Exception:
+    ROCM_AVAILABLE = False
+
+try:
+    import zentorch  # noqa: F401 # type: ignore
+
+    ZENTORCH_AVAILABLE = True
+except Exception:
+    ZENTORCH_AVAILABLE = False
+
+
+def get_amd_zentorch_env():
     """
-    An alternative to torchrun that's optimized to maximize inter-devices bandwidth.
+    A cli command that sets a couple of zentorch-optimal environment variables
+    before running the command passed to it.
 
-    Usage: amdrun --ngpus <num_gpus> <script> <script_args>
+    Usage: amdrun <script> <script_args>
+    Example: amdrun torchrun --nproc_per_node 4 train.py
     """
 
-    # 4 is the minimum number of arguments required (amdrun --ngpus <num_gpus> <script>)
-    assert len(sys.argv) >= 4, "Usage: amdrun --ngpus <num_gpus> <script> <script_args>"
+    CPU_COUNT = os.cpu_count()
+    MALLOC_CONF = "oversize_threshold:1,background_thread:true,metadata_thp:auto,dirty_decay_ms:-1,muzzy_decay_ms:-1"
+    LD_PRELOAD = "/usr/lib/x86_64-linux-gnu/libjemalloc.so"
+    GOMP_CPU_AFFINITY = f"0-{CPU_COUNT - 1}"
+    OMP_NUM_THREADS = f"{CPU_COUNT}"
+    OMP_WAIT_POLICY = "ACTIVE"
+    ZENDNN_GEMM_ALGO = "4"
+    OMP_DYNAMIC = "False"
+
```

Comment:
Will ask EPYC team for some doc/explanation of why we use these values specifically.
The only one that might need to change here is `LD_PRELOAD` because we can't be sure where the `jemalloc` binary is, I can add a check that this file exist and show a warning suggesting setting it manually (like `LD_PRELOAD=... amdrun python ...`. the others were suggested as is by the EPYC team.